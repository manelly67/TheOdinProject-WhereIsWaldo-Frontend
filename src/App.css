body {
  color: var(--color-base-text);
  background-color: var(--color-bg-first);
}
 

#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
  color: var(--color-base-text);
  background-color: var(--color-bg-first);
}

.setOfButtons {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  gap: 10px;
}

.iconImg {
  height: 30px;
  width: auto;
}


@media (max-width: 770px) {
 .bar {
  position: absolute;
  top: 4%;
  width: 300px;
  display: grid;
  grid-template-rows: 40px;
  grid-template-columns: 1fr 1fr 1fr 1fr 2fr;

 }
 .themeButton{
  grid-row: 1;
  grid-column-start: 5;
  grid-column-end: 6;
 }

.setOfLinks{
  grid-row: 1;
  grid-column-start: 1;
  grid-column-end:3;
  }

}


.setOfLinks {
  position: absolute;
  top: 15%;
  right: 3%;

  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  gap: 3px;
}

button {
  border-radius: 8px;
  border: 1px solid transparent;
  padding: 0.6em 1.2em;
  font-size: 1em;
  font-weight: 500;
  font-family: inherit;
  background-color: var(--color-bg-button);
  cursor: pointer;
  transition: border-color 0.25s;
  transition: transform 250ms ease-in-out;
}
button:hover {
  border-color: #84cc16;
  transform: scale(1.02);
}
button:focus,
button:focus-visible {
  outline: 4px auto -webkit-focus-ring-color;
}


.themeButton {
  position: absolute;
  top: 4%;
  right: 3%;
  z-index: 1;
  padding: 0.3em 0.9em;

  transition: transform 250ms ease-in-out;
}

.themeButton:hover {
  transform: scale(1.1);
 
}

.themeButton:active {
  transform: scale(1.1);
 
}
