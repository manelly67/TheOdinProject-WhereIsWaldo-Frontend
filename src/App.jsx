import { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";
import "./styles/App.css";
import ToggleTheme from "./components/ToggleTheme";

let didInit = false;

function App() {
  const titleDiv = document.querySelector("title");
  if (titleDiv) {
    titleDiv.textContent = "WHERE IS WALDO";
  }

  const navigate = useNavigate();
  const name_game_1 = "Waldo In The Galactic City";
  const name_game_2 = "Oh! Waldo is not here";
  const [playerId, setPlayerId] = useState(null);
  const [playerName, setPlayerName] = useState(null);
  const player = playerId === null ? null : { id: playerId, name: playerName };

  useEffect(() => {
    if (!didInit) {
      didInit = true;
      // âœ… Only runs once per app load
      const id = crypto.randomUUID();
      setPlayerId(id);
      setPlayerName("anonymous");
    }
  }, []);

  console.log(player);

  return (
    <>
      <ToggleTheme theme="light" />
      <h1>Where is Waldo - The Game</h1>

      <div className="setOfButtons">
        <button
          onClick={() => {
            navigate("/board", {
              replace: true,
              state: { player: player, gameName: name_game_1 },
            });
          }}
        >
          <h3>{name_game_1}</h3>
        </button>
        <button
          onClick={() => {
            navigate("/board", {
              replace: true,
              state: { player: player, gameName: name_game_2 },
            });
          }}
        >
          <h3>{name_game_2}</h3>
        </button>
      </div>
    </>
  );
}

export default App;
