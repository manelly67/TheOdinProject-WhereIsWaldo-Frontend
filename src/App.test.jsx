import React from "react";
import { MemoryRouter } from "react-router-dom";
import { describe, it, expect, vi } from "vitest";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import toHaveStyle from "@testing-library/jest-dom";
import App from "./App.jsx";


describe("App Component", () => {
  it("display the page", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();

    expect(screen.getAllByRole("button").length).toBe(3);
  });

 

});