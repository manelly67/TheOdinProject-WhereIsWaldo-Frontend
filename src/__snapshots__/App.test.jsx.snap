// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > display the page 1`] = `
<div>
  <button
    class="themeButton"
  >
    <img
      alt="theme-light-dark"
      class="iconImg"
      height="30px"
      src="/src/assets/theme-light-dark.png"
      width="30px"
    />
  </button>
  <h1>
    Where is Waldo - The Game
  </h1>
  <div
    class="setOfButtons"
  >
    <button>
      <h3>
        Waldo In The Galactic City
      </h3>
    </button>
    <button>
      <h3>
        Another image
      </h3>
    </button>
  </div>
</div>
`;
